### PR TITLE
#11Calculate the GPA and show the student name and GPA by position.

### PR DESCRIPTION
**Pull request by sanishmaharjan**
Calculate GPA #11 
Write a program that calculates percentage and GPA of 5 students.
User will enter the marks of each 5 students from the keyboard.
Each of the students has following attributes: name, class, rollnum, and marks obtained in 5 subjects.
Calculate the GPA and show the student name and GPA by position.[higher GPA should display on top]

Please code review. @bhuwang @alinashakya @krishnatimilsina @adhpawal @Satish-Gupta

### CODE REVIEW

File: src/com/lftechnology/java/training/sanish/student/CalculateGPA.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	private int studentNumber = 0;
+	private ArrayList<Student> students = new ArrayList<Student>();
+
+	public static void main(String args[]) {
+		CalculateGPA calculateGPA = new CalculateGPA();
+		try (Scanner scanner = new Scanner(System.in)) {
+			LOGGER.log(Level.INFO, "Enter number of students : ");
+			calculateGPA.setStudentNumber(UserInput.getNumber(scanner,
+					Constants.MIN_STUDENT_NUM, Constants.MAX_STUDENT_NUM));
+			calculateGPA.setStudentInformation(scanner);
```
Comment (utsabneupane) [inline src/com/lftechnology/java/training/sanish/student/CalculateGPA.java:26]: is there a reason of passing scanner object as a parameter? Scanner could have been defined static at top and could have used in whole class..

Comment (sanishmaharjan) [inline src/com/lftechnology/java/training/sanish/student/CalculateGPA.java:26]: The Scanner variable need to declared / instantiated & assigned to the variable  inside the parentheses after the try keyword. Static Scanner defined at top not supported by Try-with-resources.

File: src/com/lftechnology/java/training/sanish/student/ResultCalculation.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	/**
+	 * Return grade base on percentage supplied
+	 * 
+	 * @param percentage
+	 *            {@link Float} percentage obtain by student
+	 * @return {@link Character} grade obtain
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public static char getGrade(float percentage) {
+		if (percentage > 80) {
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/sanish/student/ResultCalculation.java]: what if my percentege is 80 then my grade 'D'

File: src/com/lftechnology/java/training/sanish/student/CalculateGPA.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	/**
+	 * Set mask of all subjects of student
+	 * 
+	 * @param scanner
+	 *            {@link Scanner}
+	 * @param student
+	 *            {@link Student} Student
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public void setStudentMasks(Scanner scanner, Student student) {
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/sanish/student/CalculateGPA.java]: masks?

File: src/com/lftechnology/java/training/sanish/student/UserInput.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public static String getName(Scanner scanner) {
+		String name;
+		Pattern pattern = Pattern.compile("[a-z A-Z]*");
+		while (!scanner.hasNext(pattern)) {
+			LOGGER.log(Level.WARNING, "Only characters are allowed : ");
+			scanner.next();
+		}
+		name = scanner.next();
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/sanish/student/UserInput.java]: you are using [a-z A-Z] which includes a space also.
SO when user enters 'sanish maharjan' then scanner.next() is called and stores only that character before space.
ie.name is assigned only (sanish) (name='sanish')
So it is better to use scanner.nextLine() here

**Pull request merged**